demo/idxd-init.sh

```diff
@@ -2,23 +2,38 @@
 
 set -euo pipefail
 
-DEV="${IDXD_DEVICE_TYPE:-dsa}"
+DEVICE_TYPE="${DEVICE_TYPE:-dsa}"
 NODE_NAME="${NODE_NAME:-}"
 
+case $DEVICE_TYPE in
+    dsa)
+	IDXD_DEVICE=$DEVICE_TYPE
+        ;;
+    iaa)
+	IDXD_DEVICE=iax
+        ;;
+    *)
+        echo "ERROR: invalid device type: $DEVICE_TYPE (supported: dsa | iaa)"
+	exit 1
+        ;;
+esac
+
+DEV="$DEVICE_TYPE"
+
 function cmd() {
 
     echo "$@"
 
     "${@}"
 }
 
-for i in $(accel-config list | jq '.[].dev' | grep "$DEV" | sed 's/\"//g'); do
+for i in $(accel-config list | jq -r '.[].dev' | grep "$IDXD_DEVICE"); do
 
     cmd accel-config disable-device "$i"
 
 done
 
-for i in $(accel-config list --idle | jq '.[].dev' | sed -ne "s/\"$DEV\([0-9]\+\)\"/\1/p"); do
+for i in $(accel-config list --idle | jq -r '.[].dev' | sed -ne "s/$IDXD_DEVICE\([0-9]\+\)/\1/p"); do
```

Comment:
I wonder if this `sed` can be run without depending on "iax". That seems to be the only dependency we have. The `grep` part could be handled using "grep dsa" and "grep not dsa". Thoughts?